projects/kubernetes/kubernetes/build/lib/binaries.sh

```diff
@@ -21,33 +21,39 @@ function build::binaries::kube_bins() {
 
   export KUBE_GIT_VERSION_FILE=$kube_git_version_file
 
-    cd $repository
-
-
-    # Ideally, to avoid checksum differences due to modules being downloaded by go mod download vs
-    # vendored in the vendor directly, which really only comes into play when build locally vs in a clean
-    # builder-base, we would run `./hack/update-vendor.sh` to redownload all the modules avoiding this difference
-    # We do this for all projects, however, with kubernetes we actually patch some files in vendor
-    # so if we ran this it would overwrite our patched changes
-    # TODO: potentially look at ways to run this but reset back to patches after downloads?
-    # or once we are using go 1.15, using the seperate cache dirs like we do could help?
-    # ./hack/update-vendor.sh
-
-    # Build all core components for linux arm64 and amd64
-    # GOLDFLASGS
-    # * strip symbol, debug, and DWARF tables
-    # * set an empty build id for reproducibility
-    # * build static binaries
-    # Run in two steps to support passing -trimpath
-    export CGO_ENABLED=0
-    export GOLDFLAGS='-s -w -buildid=""'
-    export KUBE_STATIC_OVERRIDES="cmd/kubelet \
-        cmd/kube-proxy \
-        cmd/kubeadm \
-        cmd/kubectl \
-        cmd/kube-apiserver \
-        cmd/kube-controller-manager \
-        cmd/kube-scheduler"
+  cd $repository
+  
+  
+  # kube build scripts set GOCACHE and GOMODCACHE to be in the ./_output/local/go/cache directory
+  # relative to the kubernetes source folder instead
+  # unsetting these two which we set in our common script to allow kube to fully manage these locations
+  # and keep the directories we need to clean consistent through versions
+  unset GOCACHE GOMODCACHE
```

Comment:
prior to 1.27 they always set GOCACHE in the kube bin script. In 1.27 they allow the user to override this, which we do by default in our scripts. This was making our cleanup below not complete. This feels like the easiest fix and the one that is most consistent with how we have used these kube scripts.